### ISSUE TITLE
Unable to write a unit test for Spring scheduler implemented with Shedlock

### ISSUE THREAD

**Issue by abhishekjoy1** [lukas-krecan/ShedLock#326]
**Is your feature request related to a problem? Please describe.**
I have implemented a  Spring scheduler with Shedlock as follows:

```
  @Scheduled(cron = "0 0/1 * * * *")
  @SchedulerLock(name = "gusSyncSAData",
      lockAtMostFor = "${shedlock.myScheduler.lockAtMostFor}",
      lockAtLeastFor = "${shedlock.myScheduler.lockAtLeastFor}")
  public void syncSAData() {
    //To assert that the lock is held
    LockAssert.assertLocked();
    ..
  }
```

But I am unable to write a unit test for the same because of the line `LockAssert.assertLocked.`

I have tried with the approach you suggested in the documentation:

  LockAssert.TestHelper.makeAllAssertsPass(true);

But it cannot find any symbol TestHelper, so the code is not getting compiled.
   
**Describe the solution you'd like**
It would help if you add one concrete unit test for a scheduler as an example in the document.

**Describe alternatives you've considered**
I did not find any other alternative.

**Comment by lukas-krecan** [comment]
The example is here https://github.com/lukas-krecan/ShedLock/commit/e8d63b7c56644c4189e0a8b420d8581d6eae1443

**Comment by abhishekjoy1** [comment]
Thanks a ton, @lukas-krecan for your response. I have tried as per the example, but I am getting compiler error as it is unable to find any symbol TestHelper. 

Should I include a separate dependency for unit test ?

**Comment by lukas-krecan** [comment]
It's strange, can you please paste here the compiler error message? Which ShedLock do you use?

**Comment by abhishekjoy1** [comment]
Hi @lukas-krecan , yeah I have been able to fix this by upgrading to 4.14.0. Now, this is working fine. Thanks a ton.

Though it is already mentioned in the doc that LockAssert is used for handling misconfiguration errors, I am still curious to reproduce such a scenario, when the lock is not held by a given pod but the job gets scheduled on the same pod.

If possible, could you please give a small example? Thanks.

**Comment by lukas-krecan** [comment]
I have updated the README to explain what misconfiguration errors may happen
